### PR TITLE
CL-25 Setup a mock GPS device

### PR DESCRIPTION
**Pull request by stevelw**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    "module": "commonjs",
+    "noImplicitAny": true,
+    "removeComments": true,
+    "preserveConstEnums": true,
+    "sourceMap": true,
+    "types": ["node"],
+    "outDir": "dist",
+    "resolveJsonModule": true
+  },
+  "include": ["src/**/*", "src/**/*.json"]
```
Comment (guarde) [inline tsconfig.json]: JSON file was not included into the "dist" folder

File: src/device.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import axios from "axios";
+import "dotenv/config";
+import { DeviceUpdateBody } from "./types/DeviceTypes";
+
+const network = axios.create({
+  baseURL: process.env.BACKEND_HOST,
+});
+
+export function sendCatLocation(body: DeviceUpdateBody) {
+  return network.post("/cats/location", body);
```
Comment (guarde) [inline src/device.ts]: should be requesting "/api/devices/update"

File: testData/cat_1.json
Comment (guarde) [inline testData/cat_1.json:1]: As per our backend, trackingData should contain an array of { id: "string", lat: number, lon: number  }

**Review by guarde — CHANGES_REQUESTED** [review]

**Comment by guarde** [comment]
The issue with this is that a device with, for example, ID 9000001 doesnt exist on our backend - We should rewrite this so it does a request to create a device, then sends all the data with the device_id being the result of the create request. Otherwise the backend will try to search for device with ID 9000001 and then error

**Comment by stevelw** [comment]
@jeepies that sounds like a backend issue not a mock issue? Shouldn't we just add the needed IDs to the database seeding for dev environment?

**Comment by guarde** [comment]
> @jeepies that sounds like a backend issue not a mock issue? Shouldn't we just add the needed IDs to the database seeding for dev environment?

The issue with that is all our ID's on the backend are collision resistant unique id's (cuid), not integers. I think at this point in development, it'd be easier to migrate the mock device ID's over.

**Comment by stevelw** [comment]
I've added the valid device IDs from our database to the dataset (by default only the first one is used).

When I run I get a write permissions error. I got that before but I thought you'd got it working with our backend when you last tested? Can you replicate?

**Comment by guarde** [comment]
I'm only getting an error because `accuracy` should be a number, but its defined as `"TBD"`. Do you have a screenshot of the error you're getting?

**Comment by stevelw** [comment]
image

**Comment by stevelw** [comment]
...
image

**Comment by guarde** [comment]
> image

There's your issue - Change your `.env` to `http` - We don't have a certificate for localhost!

**Comment by stevelw** [comment]
> There's your issue - Change your `.env` to `http` - We don't have a certificate for localhost!

I couldn't see you image, but the .env edit did fix it.

I've done https://github.com/clowder-cloud/backend/pull/24 to fix the accuracy data issue - we don't know it will be a number, and if we're not using it anywhere it's better to just not check it.

**Review by guarde — APPROVED** [review]

**Pull request merged**